source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/telemetry/TfsTelemetryHelper.java

```diff
@@ -44,8 +50,21 @@
             TfsTelemetryInstrumentationInfo.isDeveloperMode());
     }
 
+    public synchronized static void checkDisableTelemetryProperty() {
+        // Unfortunately, we can't push the disabled logic onto the
+        // TelemetryClient itself because the initialization code for
+        // that object simply takes too much time. So, we have to stop
+        // the sending of telemetry here in the helper class.
+
+        // If the system property exists or the env var is true, then we disable
+        // telemetry.
+        sendingTelemetryDisabled = System.getProperty(DISABLE_TELEMETRY_PROPERTY_NAME) != null
```

Comment:
One more thing. As we wish to disable telemetry only in the specific case of the TF command usage (IntelliJ plugin), Doesn't make sense to create in the base TfsTelemetryHelper class a protected getter isTelemetryDisabled() {return false;}, and override it in the CLCTelemetryHelper class?